packages/amplify_storage_s3/android/src/main/kotlin/com/amazonaws/amplify/amplify_storage_s3/AmplifyStorageOperations.kt

```diff
@@ -197,16 +184,23 @@ class AmplifyStorageOperations {
             }
         }
 
-        private fun prepareError(@NonNull flutterResult: MethodChannel.Result, @NonNull msg: String, localizedError: String?, recoverySuggestion: String?) {
-            var errorDetails = HashMap<String, Any>();
-            errorDetails.put("PLATFORM_EXCEPTIONS", mapOf(
-                    "platform" to "Android",
-                    "localizedErrorMessage" to localizedError,
-                    "recoverySuggestion" to recoverySuggestion
-            ))
+        private fun prepareError(@NonNull flutterResult: MethodChannel.Result, @NonNull e: Exception) {
+            val errorCode = "StorageException"
+            LOG.error(errorCode, e)
+            val serializedError: Map<String, Any?> = if (e is AmplifyException) {
+                ExceptionUtil.createSerializedError(e)
+            } else {
```

Comment:
I don't know if it's possible or not to get an `AmplifyException` which is not a `StorageException` but if we did, this might not be accurate. Should we have two conditions one for AmplifyException and other for StorageException?

Reply:
I think I am going to be consistent with auth and send these as generic exceptions.